test/include/MadgraphTest.h

```diff
@@ -0,0 +1,110 @@
+// Stephan Hageboeck, CERN, 12/2020
+#ifndef MADGRAPHTEST_H_
+#define MADGRAPHTEST_H_
+
+#include <array>
+#include <iomanip>
+#include <string>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+struct ReferenceData {
+  std::vector< std::vector<std::array<double, 4>> > momenta;
+  std::vector<double> MEs;
+};
+
+std::map<unsigned int, ReferenceData> readReferenceData(const std::string& refFileName);
+
+/**
+ * Test driver providing a common interface for testing different implementations.
+ * Users need to implement:
+ * - Functions to retrieve matrix element and 4-momenta. These are used in the tests.
+ * - Driver functions that run the madgraph workflow.
+ *
+ * Usage:
+ * ```
+ * class TestImplementation : public BaseTest {
+ *   <implement functions>
+ * }
+ *
+ * TEST_F(TestImplementation, <testName>) {
+ *   <test code>
+ * }
+ */
+template<typename FpType>
+class TestDriverBase {
+ public:
+  using fptype = FpType;
+  unsigned int nparticle = 4;
+  static constexpr unsigned int np4 = 4;
+  static constexpr unsigned int niter = 2;
+  static constexpr unsigned int gpublocks = 2;
```

Comment:
For those variables it could be nice to get them out of the source and into some "config file", maybe also make a distinction between CPU and GPU execution? It's fine for me to create an issue for that and merge as such though.